Objective-C/CBLArray.m

```diff
@@ -42,89 +42,104 @@ - (instancetype) initWithArray: (NSArray*)array {
 }
 
 
-- /*internal */ (instancetype) initWithFleeceData: (CBLFLArray*)data {
-    self = [super initWithFleeceData: data];
-    if (self ) {
-        _array = [NSMutableArray array];
-        [self loadBackingFleeceData];
-    }
-    return self;
-}
-
-
 #pragma mark - GETTER
 
 
 - (nullable id) objectAtIndex: (NSUInteger)index {
+    if (!_array) {
+        id value = [super objectAtIndex: index];
+        if ([value isKindOfClass: [CBLReadOnlyDictionary class]] ||
+            [value isKindOfClass: [CBLReadOnlyArray class]]) {
+            [self copyFleeceData];
```

Comment:
I guess CBLDictionary/CBLArray could be instantiated with NSDictionary/NSArray from ReadOnlyDictionary/ReadOnlyArray. This could be cheaper than calling copyFleeceData.
However once loaded into `NSMutableArray`, following data access will be cheaper. What do you think?